### PR TITLE
ui: Update volume label validation for kubernetes

### LINKED ISSUE #3203: UI: Validate labels names for volumes
**Component**:

UI

**What happened**:

When creating a volume using specials characters in the UI is authorised for labels but denied when submitting the query to kubernetes API.

Kubernetes API answered with this message : `Volume.storage.metalk8s.scality.com "test" is invalid: metadata.labels: Invalid value: "'world'": a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')` but from the UI point of view we just see a generic error :

image

**What was expected**:

Field should be validated when they are being filled in or we should display the error message so that the user can understand that the issue comes from invalid characters.

**Steps to reproduce**

Create a volume and use a `'` as a label name or value.

### PR DESCRIPTION
**Pull request by theocerutti**
**Component**: CreateVolume

**Context**: Add validation message to labelValue input and labelName input.

**Preview**:

image (5)
image (4)

Translations:
* FR: Le label doit être composé de lettres, de chiffres ou de tiret et doit commencer et finir avec une lettre ou un chiffre.
* EN: Label must be letters, numbers or hyphens and must begin and end with a letter or a number.

Feel free to ask me to change the translation.

Closes: #3203

### CODE REVIEW

File: ui/src/translations/en.json [hunk 1]
```diff
@@ -91,6 +91,7 @@
   "volume_creation_success": "Volume {name} is being created.",
   "volume_creation_failed": "Volume {name} creation has failed.",
   "name_error": "Name must be lowercase letters, numbers and hyphens.",
+  "label_error": "Label must be letters, numbers ou hyphens and must begin and end with a letter or a number.",
```
Comment (alexandre-allard) [inline ui/src/translations/en.json]:
```suggestion
  "label_error": "Label must be letters, numbers or hyphens and must begin and end with a letter or a number.",
```

Comment (JBWatenbergScality) [inline ui/src/translations/en.json]:
```suggestion
  "label_error": "Label must be letters, numbers or hyphens and must begin and end with a letter or a number.",
```

File: ui/src/translations/en.json [hunk 2]
```diff
@@ -91,6 +91,7 @@
   "volume_creation_success": "Volume {name} is being created.",
   "volume_creation_failed": "Volume {name} creation has failed.",
   "name_error": "Name must be lowercase letters, numbers and hyphens.",
+  "label_error": "Label must be letters, numbers or hyphens and must begin and end with a letter or a number.",
```
Comment (alexandre-allard) [inline ui/src/translations/en.json]: That's not totally accurate, description from K8s:
```
A valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')
```
We're missing dot `.` and underscore `_`

File: ui/src/containers/CreateVolume.js [hunk 3]
```diff
@@ -315,6 +314,7 @@ const CreateVolume = (props) => {
   };
 
   const volumeNameRegex = /^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$/;
+  const labelNameRegex = /^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?$/;
```
Comment (JBWatenbergScality) [inline ui/src/containers/CreateVolume.js]: This is a valid regex for label values, for keys the validation is a bit more complex, please refer to https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set

Comment (JBWatenbergScality) [reply ui/src/containers/CreateVolume.js]: We might need to change the error message to follow this

**Review by JBWatenbergScality — CHANGES_REQUESTED** [review]

File: ui/src/translations/en.json (hunk 2 above)
Comment (theocerutti) [reply ui/src/translations/en.json]: Should I change it to: "Label must be letters, numbers, hyphens, dots or underscores and must begin and end with a letter or a number." ?
Or like the example: "A valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345')"

Comment (alexandre-allard) [reply ui/src/translations/en.json]: As you prefer, both are correct.

**Review by ChengYanJin — APPROVED** [review]

**Pull request merged**